### DISCUSSION TITLE
Usage of G28

### DISCUSSION THREAD

**Discussion by khturnings** [winder/Universal-G-Code-Sender#1975] [General]
It is a known issue/feature that GCode files with G28 in them can cause some issues with unexpected gantry movement especially with those new to CNC like me. My gantry slammed into my end stops and tried to keep going! 

According to http://linuxcnc.org/docs/html/gcode/g-code.html#gcode:g28-g28.1 info on G28 is stored in parameters 5161-5166 and is set by using G28.1

Can an error check be done when loading the GCode file to find out if these parameter values are set (i.e. if they have never been set they will be a default value) or if they are outside the machine size limits, then give a warning or offer to fix the GCode file and reload or something like that?

**Comment by ElsGear** [comment]
I've got the same problem and made an electronic AutoSquare module that has to work with the G28 command to stop when reaching limit switches for that direction.
Every UGS G28 command should work correct for not slamming the gantry if I want to use the electronic AutoSquare.
